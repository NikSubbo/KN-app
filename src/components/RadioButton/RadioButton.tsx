import { FC } from 'react';
import cx from 'classnames';

export type Props = {
  option: string | null;
  checked?: boolean;
  className?: string;
};

export const RadioButton: FC<Props> = ({ option, checked, className }) => {
  if (!option) return null;

  return (
    <div className={cx(className, 'inline-flex items-center cursor-pointer')}>
      <input
        type="radio"
        id={option}
        name="radioBtn"
        defaultChecked={checked}
        className="w-4 h-4 mr-2 peer cursor-pointer"
      />
      <label
        htmlFor={option}
        className="font-medium text-sm text-darkGray peer-checked:text-black cursor-pointer"
      >
        {option}
      </label>
    </div>
  );
};

export default RadioButton;
