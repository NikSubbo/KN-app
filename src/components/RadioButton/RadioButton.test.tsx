import { render } from '@testing-library/react';
import RadioButton, { Props } from './RadioButton';

const DEFAULT_PROPS = {
  option: 'option'
};

const renderContainer = (overrideProps?: Partial<Props>) => {
  return render(<RadioButton {...DEFAULT_PROPS} {...overrideProps} />);
};

describe('RadioButton', () => {
  it('should match the snapshot', () => {
    const { container } = renderContainer();
    expect(container).toMatchSnapshot();
  });

  it('should return null for null option', () => {
    const { container } = renderContainer({
      option: null
    });
    expect(container.firstChild).toBeNull();
  });

  it('should be checked', () => {
    const { getByRole } = renderContainer({
      checked: true
    });
    expect(getByRole('radio')).toBeChecked();
  });
});
