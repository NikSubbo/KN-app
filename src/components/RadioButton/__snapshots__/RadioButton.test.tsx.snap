// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`RadioButton should match the snapshot 1`] = `
<div>
  <div
    class="inline-flex items-center cursor-pointer"
  >
    <input
      class="w-4 h-4 mr-2 peer cursor-pointer"
      id="option"
      name="radioBtn"
      type="radio"
    />
    <label
      class="font-medium text-sm text-darkGray peer-checked:text-black cursor-pointer"
      for="option"
    >
      option
    </label>
  </div>
</div>
`;
